### ISSUE TITLE
Await async function calls

### ISSUE THREAD

**Issue by Corniel** [SonarSource/sonar-csharp#1197]
### Description
When calling async methods, it seen as bad practise (for good reasons) to not to await those methods, or use the Result property of the Task (as that can lead to deadlocks).

There are basically two ways to fix this issue:
1. Propagate the async behaviour to the calling method (not always possible)
2. Use Task.Run().Result or Task.Run().Wait() or Task.Run().GetAwaiter().GetResult()

### Repro steps

``` C#
using System.Threading.Tasks;

namespace SonarAnalyzer.UnitTest.TestCases
{
    public class AwaitAsyncFunctionCall
    {
        public boo Run()
        {
            SomeAction(); // Not compliant
            return SomeFunctionAsync().Result; // Not compliant
        }

        public bool RunSync()
        {
            Task.Run(() => SomeAction()).Wait(); // Compliant
            return Task.Run(() => SomeFunctionAsync()).Result; // compliant
        }

        public async Task<bool> RunAsync()
        {
            await SomeAction().ConfigureAwait(false); // Compliant
            return await SomeFunctionAsync().ConfigureAwait(false); // Compliant
        }

        public Task<bool> SomeFunctionAsync() => Task.FromResult(true);

        public Task SomeAction() => Task.CompletedTask;
    }
}
```

It would be nice if Sonar could provide a rule that helps developers doing it right.

### Related information

https://cpratt.co/async-tips-tricks/

**Comment by Evangelink** [comment]
Hi @Corniel,

Thanks for the suggestion. I think this will be covered by #1159 (see the RSPEC linked in my comment).

**Comment by Corniel** [comment]
I think so too. I tried to find if there was already work done or planned, but couldn't find it. You mark this as a duplicate I assume.

**Comment by Evangelink** [comment]
Duplicate of #1159
